include/sys/mmp.h

```diff
@@ -58,6 +59,20 @@ extern ulong_t zfs_multihost_interval;
 extern uint_t zfs_multihost_fail_intervals;
 extern uint_t zfs_multihost_import_intervals;
 
+/*
+ * If MMP is unable to find a leaf vdev to write an MMP block to,
+ * it will set *error to one of the below error codes.
+ *
+ * MMP_WRITE_PENDING	At least one writeable leaf vdev was found, but it had a
+ * 			pending MMP write.
+ * MMP_TREE_RO		At least one leaf was found, but none were writeable.
+ */
+typedef enum mmp_error {
+	MMP_WRITE_TREE_RO = ENETDOWN,
```

Comment:
Tethering these to `EHWPOISON + 1` might  be a better choice since then they won't overlap with any legitimate existing `errno` for the architecture.  Mind you this value will vary between architectures but that's also true for `ENETDOWN`.  Alternately, you could start they at say `500` which is above most values I'm aware of.

Reply:
Switched to returning bitmap instead as discussed.